web-admin/src/features/projects/status/ProjectResources.svelte

```diff
@@ -2,54 +2,154 @@
+  import { eventBus } from "@rilldata/web-common/lib/event-bus/event-bus";
+  import { useResources } from "./selectors";
+  import { ResourceKind } from "@rilldata/web-common/features/entity-management/resource-selectors";
+  import { onNavigate } from "$app/navigation";
+  import { onMount } from "svelte";
 
   const queryClient = useQueryClient();
   const createTrigger = createRuntimeServiceCreateTrigger();
 
+  let isConfirmDialogOpen = false;
+  let isPollingEnabled = false;
+  let currentResourceName: string | undefined;
   let isReconciling = false;
-  let isRefreshConfirmDialogOpen = false;
+  let isLoaded = false;
+
+  const INITIAL_POLL_INTERVAL = 1_000;
+  const MAX_POLL_INTERVAL = 10_000;
+  const BACKOFF_THRESHOLD_MS = 30_000;
+  let pollStartTime: number | null = null;
 
   $: ({ instanceId } = $runtime);
 
-  $: resources = createRuntimeServiceListResources(
-    instanceId,
-    // All resource "kinds"
-    undefined,
-    {
-      query: {
-        select: (data) => {
-          // Filter out the "ProjectParser" resource
-          return data.resources.filter(
-            (resource) =>
-              resource.meta.name.kind !== "rill.runtime.v1.ProjectParser",
-          );
-        },
-        refetchOnMount: true,
-        refetchOnWindowFocus: true,
-        refetchInterval: isReconciling ? 500 : false,
-      },
-    },
-  );
+  function isResourceErrored(resource: V1Resource) {
+    return resource.meta.reconcileError;
+  }
 
-  $: isAnySourceOrModelReconciling = $resources?.data?.some(
-    (resource) =>
+  function isResourceReconciling(resource: V1Resource) {
+    return (
       resource.meta.reconcileStatus ===
         V1ReconcileStatus.RECONCILE_STATUS_PENDING ||
       resource.meta.reconcileStatus ===
-        V1ReconcileStatus.RECONCILE_STATUS_RUNNING,
+        V1ReconcileStatus.RECONCILE_STATUS_RUNNING
+    );
+  }
+
+  $: resources = useResources(instanceId, {
+    select: (data: V1ListResourcesResponse) => ({
+      ...data,
+      // Filter out project parser and refresh triggers
+      resources: data?.resources?.filter(
+        (resource: V1Resource) =>
+          resource.meta.name.kind !== ResourceKind.ProjectParser &&
+          resource.meta.name.kind !== ResourceKind.RefreshTrigger,
+      ),
+    }),
+    refetchInterval: (data) => {
+      if (
+        !isPollingEnabled ||
+        $resources?.isError ||
+        data?.resources?.some(isResourceErrored)
+      ) {
+        pollStartTime = null;
+        return false;
+      }
+
+      // Initialize poll start time if not set
+      if (!pollStartTime) {
+        pollStartTime = Date.now();
+      }
+
+      // Calculate time elapsed since polling started
+      const elapsedTime = Date.now() - pollStartTime;
+
+      // After threshold, gradually increase interval to MAX_POLL_INTERVAL
+      if (elapsedTime > BACKOFF_THRESHOLD_MS) {
+        return MAX_POLL_INTERVAL;
+      }
+
+      return INITIAL_POLL_INTERVAL;
```

Comment:
Nit: can we do more graceful scaling? Firing 30 requests in 30 seconds seems a little heavy.